### PR TITLE
feat: extract statetest models/structs to standalone crate

### PR DESCRIPTION
**Pull request by royvardhan**
Closes: #1787

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,14 +18,15 @@ members = [
...
+    "crates/statetest-types",
 
     # examples
     "examples/block_traces",
     "examples/contract_deployment",
     "examples/database_components",
     "examples/database_ref",
     "examples/uniswap_get_reserves",
-    "examples/uniswap_v2_usdc_swap",
+    "examples/uniswap_v2_usdc_swap",
```
Comment (rakita) [inline Cargo.toml:29]:
```suggestion
    "examples/uniswap_v2_usdc_swap",
```

File: crates/statetest-types/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+unreachable_pub = "warn"
+unused_must_use = "deny"
+rust_2018_idioms = "deny"
+
+[lints.rustdoc]
+all = "warn"
+
+[dependencies]
+# revm
+revm = { workspace = true }
```
Comment (rakita) [inline crates/statetest-types/Cargo.toml:26]: would be good to include `std` and `serde` feature
```suggestion
revm = { workspace = true, features = ["std", "serde"] }
```

**Review by rakita — APPROVED** [review]
Cool! lgtm with one nit.

**Comment by royvardhan** [comment]
> Cool! lgtm with one nit.

Done! Ci green.

**Pull request merged**